File: tests/csharp_sample_project/Item.cs
// tests/sample_project2/Item.cs

namespace SampleProject2;

/// <summary>
/// Represents a single data item to be processed.
/// </summary>
public class Item
{
    public int ItemId { get; set; }
    public string Name { get; set; }
    public double Value { get; set; }
    public bool Processed { get; private set; }

    /// <summary>
    /// Initializes a new instance of the Item class.
    /// </summary>
    public Item(int itemId, string name, double value)
    {
        ItemId = itemId;
        Name = name;
        Value = value;
        Processed = false; // Default value
    }

    /// <summary>
    /// Sets the processed flag to true, updating the item's state.
    /// </summary>
    public void MarkAsProcessed()
    {
        Console.WriteLine($"Model Item {ItemId}: Marking '{Name}' as processed.");
        this.Processed = true;
    }

    /// <summary>
    /// Returns a user-friendly string representation of the item.
    /// </summary>
    public override string ToString()
    {
        string status = Processed ? "Processed" : "Pending";
        return $"Item(ID={ItemId}, Name='{Name}', Value={Value:F2}, Status={status})";
    }
}

File: tests/swift_sample_project/SampleProject2/Item.swift
// Sources/SampleProject2/Item.swift

import Foundation

/// Represents a single data item to be processed.
public class Item: CustomStringConvertible {
    public let itemId: Int
    public let name: String
    public let value: Double
    public var processed: Bool

    /// Initializes a new instance of the Item class.
    public init(itemId: Int, name: String, value: Double, processed: Bool = false) {
        self.itemId = itemId
        self.name = name
        self.value = value
        self.processed = processed
    }

    /// Sets the processed flag to true, updating the item's state.
    public func markAsProcessed() {
        print("Model Item \(self.itemId): Marking '\(self.name)' as processed.")
        self.processed = true
    }

    /// Returns a user-friendly string representation of the item.
    public var description: String {
        let status = processed ? "Processed" : "Pending"
        return "Item(ID=\(itemId), Name='\(name)', Value=\(String(format: "%.2f", value)), Status=\(status))"
    }
}